api/spec/serializers/spree/api/v2/storefront/icon_serializer_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+describe Spree::V2::Storefront::IconSerializer do
+  subject { described_class.new(icon) }
+
+  let(:menu) { create(:menu) }
+  let(:menu_item) { create(:menu_item, menu: menu) }
+  let(:icon) { create(:icon, viewable: menu_item) }
+
+  it { expect(subject.serializable_hash).to be_kind_of(Hash) }
+
+  it do
+    expect(subject.serializable_hash).to eq(
+      {
+        data: {
+          id: icon.id.to_s,
+          type: :icon,
+          attributes: {
+            url: Rails.application.routes.url_helpers.polymorphic_url(icon.attachment, only_path: true),
+            viewable_id: icon.viewable_id,
+            viewable_type: icon.viewable_type
```

Comment:
@MatthewKennedy we don't need viewable_id/viewable_type attributes

Reply:
@damianlegawiec is this just on the storefront or remove these attributes on platform API also?